src/pilot/assign_role.py

```diff
@@ -1014,34 +1026,58 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
 def get_drives(drac_client):
     spinners = []
     ssds = []
-    physical_disks = drac_client.list_physical_disks()
-    for physical_disk in physical_disks:
-        # Eliminate physical disks that in a state other than non-RAID
-        if physical_disk.raid_status != "non-RAID":
-            LOG.info("Skipping disk {id}, because it has a RAID status of "
-                     "{raid_status}".format(
-                         id=physical_disk.id,
-                         raid_status=physical_disk.raid_status))
-            continue
+    virtual_disks = drac_client.list_virtual_disks()
 
-        # Eliminate physical disks that have an error status
-        if physical_disk.status == 'error':
-            LOG.warning("Not using disk {id}, because it has a status of "
-                        "{status}".format(id=physical_disk.id,
-                                          status=physical_disk.status))
-            continue
+    raid0_disks = [vd for vd in virtual_disks if vd.raid_level != '1']
+
+    raid1_disks = [vd.physical_disks for vd in virtual_disks
+                   if vd.raid_level == '1'][0]
```

Comment:
I don't think the above 2 lines are right.  It looks like it will make a list of lists of physical drives, and then take the first list of physical drives for the first RAID1.  Since we don't know which RAID1 will be used for the OS drive, we need raid1_disks to be a list of all physical disks for all RAID1s.  I think you want to do something like this:
raid1_disks=[]
for vd in virtual_disks:
  if vd.raid_level == '1':
    raid1_disks.extend(vd.physical_disks)

Reply:
Done